docs: add download page (#10271)
apache/datafusion: docs/source/download.md

# Download

The official Apache DataFusion releases are provided as source artifacts.

## Releases

> The latest source release is [0.45.0](https://www.apache.org/dyn/closer.lua/opendal/0.45.0/apache-opendal-0.45.0-src.tar.gz?action=download) ([asc](https://downloads.apache.org/opendal/0.45.0/apache-opendal-0.45.0-src.tar.gz.asc),

Author: Once we make the first release, we can update these links and merge this patch.

Author: <img width="1728" alt="image" src="https://github.com/apache/datafusion/assets/18818196/8c323379-e2da-40ac-8f8d-3d6abce636a5">

<!---
  Licensed to the Apache Software Foundation (ASF) under one
  or more contributor license agreements.  See the NOTICE file
  distributed with this work for additional information
  regarding copyright ownership.  The ASF licenses this file
  to you under the Apache License, Version 2.0 (the
  "License"); you may not use this file except in compliance
  with the License.  You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

  Unless required by applicable law or agreed to in writing,
  software distributed under the License is distributed on an
  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
  KIND, either express or implied.  See the License for the
  specific language governing permissions and limitations
  under the License.
-->

# Download

The official Apache DataFusion releases are provided as source artifacts.

## Releases

> The latest source release is [0.45.0](https://www.apache.org/dyn/closer.lua/opendal/0.45.0/apache-opendal-0.45.0-src.tar.gz?action=download) ([asc](https://downloads.apache.org/opendal/0.45.0/apache-opendal-0.45.0-src.tar.gz.asc),

Reviewer: I assume the opendal references need replacing with datafusion?

Reviewer: Thanks, I missed that. We are planning on creating a new release this week. I am going to start the process tomorrow.

https://github.com/apache/datafusion/issues/10217

Reviewer: We could link to the last release in the arrow site for now?

https://downloads.apache.org/arrow/arrow-datafusion-37.0.0/

<!---
  Licensed to the Apache Software Foundation (ASF) under one
  or more contributor license agreements.  See the NOTICE file
  distributed with this work for additional information
  regarding copyright ownership.  The ASF licenses this file
  to you under the Apache License, Version 2.0 (the
  "License"); you may not use this file except in compliance
  with the License.  You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

  Unless required by applicable law or agreed to in writing,
  software distributed under the License is distributed on an
  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
  KIND, either express or implied.  See the License for the
  specific language governing permissions and limitations
  under the License.
-->

# Download

> The official Apache DataFusion releases are provided as source artifacts.

Reviewer 2: I think we may cause confusion if we don't also clarify how the code on crates.io related to the downloads linked from this page

```suggestion
While DataFusion is also distributed via the Rust [crates.io] package manager as a convenience, the 
official Apache DataFusion releases are provided as source artifacts. 

[crates.io]: https://crates.io/crates/datafusion
```

## Releases

The latest source release is [37.0.0][source-link] ([asc][asc-link],
[sha512][sha512-link]).

[source-link]: https://www.apache.org/dyn/closer.lua/arrow/arrow-datafusion-37.0.0/apache-arrow-datafusion-37.0.0.tar.gz?action=download
[asc-link]: https://www.apache.org/dyn/closer.lua/arrow/arrow-datafusion-37.0.0/apache-arrow-datafusion-37.0.0.tar.gz.asc?action=download

> [sha512-link]: https://www.apache.org/dyn/closer.lua/arrow/arrow-datafusion-37.0.0/apache-arrow-datafusion-37.0.0.tar.gz.sha512?action=download

Author: Thank you! I often forget the correct links in use ... 😢

> For older releases, please check the [archive](https://archive.apache.org/dist/datafusion/).

Reviewer 3: ```suggestion
For previous releases, please check the [archive](https://archive.apache.org/dist/datafusion/).
```

> For even older releases when DataFusion is a subproject of Apache Arrow, please check [Arrow's archive](https://archive.apache.org/dist/arrow/).

Reviewer 3: ```suggestion
For older releases (<= 37) when DataFusion was a subproject of Apache Arrow, please check [Arrow's archive](https://archive.apache.org/dist/arrow/).
```

Minor rewording

> For older releases (<= 37) when DataFusion was a subproject of Apache Arrow, please check [Arrow's archive](https://archive.apache.org/dist/arrow/).

Reviewer 4: ```suggestion
Note: For releases earlier than 37.0.0 please check [Arrow's archive](https://archive.apache.org/dist/arrow/).
```

Author: ```suggestion
For releases earlier than 37.0.0, please check [Arrow's archive](https://archive.apache.org/dist/arrow/).
```
